src/chain/validators-impl.ts

```diff
@@ -86,8 +86,10 @@ export class ValidatorsImpl<Chain> implements Validators<Chain> {
   equals(comparison: string) {
     return this.addStandardValidation(validator.equals, comparison);
   }
-  isAfter(date?: string) {
-    return this.addStandardValidation(validator.isAfter, date);
+  isAfter(date?: string): Chain;
+  isAfter(options?: Options.IsAfterOptions): Chain;
+  isAfter(dateOrOptions?: string | Options.IsAfterOptions) {
+    return this.addStandardValidation(validator.isAfter, dateOrOptions);
```

Comment:
This (and `isISBN`) could/should be a single overload, with a parameter of union type instead.

From [TS handbook](https://www.typescriptlang.org/docs/handbook/2/functions.html#function-overloads):

> we can invoke it with strings or arrays. However, we can’t invoke it with a value that might be a string or an array, because TypeScript can only resolve a function call to a single overload:
> ...
> Always prefer parameters with union types instead of overloads when possible

I think that because of the above, usage might become picky for nothing:

```js
typeof date === 'string' ? chain.isAfter(date) : chain.isAfter(date)
```